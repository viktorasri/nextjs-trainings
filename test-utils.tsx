import { ThemeProvider } from '@emotion/react';
import { render, RenderOptions } from '@testing-library/react';
import { FC, PropsWithChildren, ReactElement } from 'react';
import { Themes } from './styles/themes';

const Wrapper: FC<PropsWithChildren> = ({ children }) => (
  <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
);

const customRender = (ui: ReactElement, options?: RenderOptions) => {
  return render(ui, { wrapper: Wrapper, ...options });
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
