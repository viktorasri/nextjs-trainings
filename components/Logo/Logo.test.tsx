import { render } from '~/test-utils';
import { Logo } from './Logo';

describe('Logo test cases', () => {
  it('Render check', () => {
    const { asFragment } = render(<Logo>Logo</Logo>);
    expect(asFragment()).toMatchSnapshot();
  });
  it('Render with custom size', () => {
    const { asFragment } = render(<Logo size={10}>Logo</Logo>);
    expect(asFragment()).toMatchSnapshot();
  });
});
