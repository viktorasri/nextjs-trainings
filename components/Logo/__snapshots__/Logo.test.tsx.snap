// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo test cases Render check 1`] = `
<DocumentFragment>
  <header
    class="css-q19ajl"
  >
    Logo
  </header>
</DocumentFragment>
`;

exports[`Logo test cases Render with custom size 1`] = `
<DocumentFragment>
  <header
    class="css-1552zee"
    size="10"
  >
    Logo
  </header>
</DocumentFragment>
`;
